internal/controller/comparators/authpolicy_comparator_test.go

```diff
@@ -0,0 +1,107 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package comparators_test
+
+import (
+	"testing"
+
+	"github.com/opendatahub-io/odh-model-controller/internal/controller/comparators"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func createAuthPolicy(appLabel, strategy string) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "kuadrant.io/v1",
+			"kind":       "AuthPolicy",
+			"metadata": map[string]interface{}{
+				"name":      "test-authn",
+				"namespace": "test-ns",
+				"labels": map[string]interface{}{
+					"app": appLabel,
+				},
+			},
+			"spec": map[string]interface{}{
+				"defaults": map[string]interface{}{
+					"strategy": strategy,
+				},
+			},
+		},
+	}
+}
+
+func createAuthPolicyWithoutLabels(strategy string) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
```

Comment:
To avoid duplication, move the common part to an env and add the labels in the function that needs it.

Also, isn't it better to modify the caller to add a flag to instruct whether or not to add the labels?